src/Google.Pubsub.V1/SubscriberClient.cs

```diff
@@ -0,0 +1,661 @@
+        /// </summary>
+        /// <param name="grpcClient">A GRPC client to wrap.</param>
+        /// <param name="settings">
+        /// An optional <see cref="SubscriberSettings"/> to configure this wrapper.
+        /// If null or not specified, then the default settings are used.
+        /// </param>
+        /// <returns>A <see cref="SubscriberSettings"/> that wraps the specified GRPC client.</returns>
+        public static SubscriberClient ToClient(
+            this Subscriber.ISubscriberClient grpcClient,
+            SubscriberSettings settings = null
+        ) => new SubscriberClientImpl(grpcClient, settings);
+    }
+
+    /// <summary>
+    /// Settings for a Subscriber wrapper.
+    /// </summary>
+    public sealed partial class SubscriberSettings : ServiceSettingsBase
+    {
+        /// <summary>
+        /// Get a new instance of the default <see cref="PublisherSettings">.
+        /// </summary>
+        /// <returns>A new instance of the default SubscriberSettings.</returns>
+        public static SubscriberSettings GetDefault() => new SubscriberSettings();
+
+        /// <summary>
+        /// Creates a deep clone of this object, with all the same property values.
+        /// </summary>
+        /// <returns>A deep clone of this set of Subscriber settings.</returns>
+        public SubscriberSettings Clone() => CloneInto(new SubscriberSettings
+        {
+        });
+    }
+
+    /// <summary>
+    /// Subscriber client wrapper, for convinient use.
+    /// </summary>
+    public abstract partial class SubscriberClient
+    {
+        /// <summary>
+        /// Default service parameters for the Subscriber service.
+        /// </summary>
+        public static class ServiceDefaults
+        {
+            /// <summary>
+            /// The default Subscriber network host.
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber network host is "pubsub-experimental.googleapis.com".
+            /// </remarks>
+            public static string Host { get; } = "pubsub-experimental.googleapis.com";
+
+            /// <summary>
+            /// The default Subscriber network port.
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber network port is 443.
+            /// </remarks>
+            public static int Port { get; } = 443;
+
+            /// <summary>
+            /// The default Subscriber scopes
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber scopes are:
+            /// <list type="bullet">
+            /// <item><description>"https://www.googleapis.com/auth/pubsub"</description></item>
+            /// <item><description>"https://www.googleapis.com/auth/cloud-platform"</description></item>
+            /// </list>
+            /// </remarks>
+            public static ReadOnlyCollection<string> Scopes { get; } = new ReadOnlyCollection<string>(new[] {
```

Comment:
Done for the type. Concrete type is still a ReadOnlyCollection<string>.
(Slightly odd naming, that ReadOnlyCollection implements all the IList things)